src/use-auto-save.test.js

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import { render, fireEvent, waitForElement } from "react-testing-library";
+import useAutoSave from "./use-auto-save";
+
+// ---------------------------
+// Mocks
+// ---------------------------
+
+const MockSaveComponent = ({ value, save, delay }) => {
+  useAutoSave(value, save, delay);
+
+  return <div>{value}</div>;
+};
+
+const MockCleanup = ({ value, save, delay }) => {
+  const [show, setShow] = useState(true);
+
+  return (
+    <>
+      {show ? (
+        <MockSaveComponent value={value} save={save} delay={delay} />
+      ) : null}
+      <button onClick={() => setShow(!show)}>Toggle AutoSave</button>
+    </>
+  );
+};
+
+// ---------------------------
+// Tests
+// ---------------------------
+
+describe("useAutoSave", () => {
+  it("runs save only once in delay period", async () => {
+    const spy = jest.fn();
+
+    const { rerender } = render(
+      <MockSaveComponent save={spy} value={1} delay={100} />
+    );
+    rerender(<MockSaveComponent save={spy} value={2} delay={100} />);
+    rerender(<MockSaveComponent save={spy} value={3} delay={100} />);
+    rerender(<MockSaveComponent save={spy} value={4} delay={100} />);
+
+    await delay(100);
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenLastCalledWith(4);
+  });
```

Comment:
I have a suspicion that we are getting a false positive here. Would you mind making sure that `MockSaveComponent` is not being re-mounted every time we call `rerender`?

Reply:
After looking at the docs for `react-testing-library` I don't think I'm right. But it would at least be worth verifying.

Reply:
Confirmed, `renrender` does not unmount the component.

Another piece, if it did remount, I think the assertions would fail because `useAutoSave` only runs the save function when the initial value changes.